settings.php

```diff
@@ -0,0 +1,37 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Implementaton of the quizaccess_honestycheck plugin.
+ *
+ * @package   quizaccess_honestycheck
+ * @author     Sumaiya Javed <sumaiya.javed@catalyst.net.nz>
+ * @copyright 2011 The Open University
+ * @license   http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+defined('MOODLE_INTERNAL') || die;
+
+$settings->add(new admin_setting_configselect(
+    'quizaccess_honestycheck/honestycheckrequired',
+    get_string('honestycheckrequired', 'quizaccess_honestycheck'),
+    get_string('honestycheckrequired', 'quizaccess_honestycheck'),
+    1,
+    array(
+        0 => get_string('notrequired', 'quizaccess_honestycheck'),
+        1 => get_string('honestycheckrequiredoption', 'quizaccess_honestycheck')
+    )
+));
```

Comment:
We should be consistent with what is done for all the other quiz settings, and give the admin control over whether the setting is advanced or locked using 

```php
    $setting->set_advanced_flag_options(admin_setting_flag::ENABLED, false);
    $setting->set_locked_flag_options(admin_setting_flag::ENABLED, false);
```

(Look at mod/quiz/settings.php if you don't understand what i mean.)

Reply:
I have fixed this as well